spec/acceptance/full_spec.rb

```diff
@@ -0,0 +1,93 @@
+require 'spec_helper_acceptance'
+
+describe 'basic installation' do
+  certdir = '/etc/pulpcore-certs'
+
+  let(:pp) {
+    <<-PUPPET
+    if $facts['os']['release']['major'] == '7' {
+      class { 'postgresql::globals':
+        version              => '12',
+        client_package_name  => 'rh-postgresql12-postgresql-syspaths',
+        server_package_name  => 'rh-postgresql12-postgresql-server-syspaths',
+        contrib_package_name => 'rh-postgresql12-postgresql-contrib-syspaths',
+        service_name         => 'postgresql',
+        datadir              => '/var/lib/pgsql/data',
+        confdir              => '/var/lib/pgsql/data',
+        bindir               => '/usr/bin',
+      }
+      class { 'redis::globals':
+        scl => 'rh-redis5',
+      }
+    }
+
+    class { 'pulpcore':
+      remote_user_environ_name => 'HTTP_REMOTE_USER',
+      apache_https_cert        => '#{certdir}/ca-cert.pem',
+      apache_https_key         => '#{certdir}/ca-key.pem',
+      apache_https_ca          => '#{certdir}/ca-cert.pem',
```

Comment:
If this is the CA that signed the cert/key pair for Apache, shouldn't the option being passed in be `apache_https_chain` ?

Reply:
The CA can be used to identify client certs. Since we need client certs, we must set that option. I think chain is optional if it's the same as the CA but can be passed in if you want to allow client certs from one CA and serve a certificate from a different CA.

Reply:
This probably does work as your tests are showing. If I go by the Apache docs:

```
 This can be used alternatively and/or additionally to SSLCACertificatePath.
```

This makes me think, for completeness of examples we should opt to specify both since it's quite confusing and the param documentation would imply that `apache_https_chain` should be set alongside cert and key. This also, as I understand it, ensures that the client will get the CA certificate to ensure trust on their end.

This works and its just for testing, but I do like when examples in testing show the best picture we can for how to use it and especially with SSL which can get confusing. So, I would prefer we specify both from my reading but wont block on it.

Reply:
I'm considering a real test of it: generate a cert to serve and a cert to auth. Verify that end to end.

Another thing to consider is that chain is deprecated. I think they want a full chain in the cert file but Apache on EL7 doesn't support that because it's too old.

However, my first goal is to even get the basic implementation for us to work. Right now I'm happy that I can test the client auth at all.